java/src/shared/models/Map.java

```diff
@@ -95,21 +173,27 @@ public boolean canSoldier(Soldier_Input params)
 	 */
 	public boolean canBuildRoad(BuildRoad_Input params)
 	{
+		boolean can = false;
 		if(roads.containsKey(params.getRoadLocation())) //already a road here man
 		{
 			return false;
 		}
+		else if(!roadIsOnMap(params.getRoadLocation()))
+		{
+			return false;
+		}
+		
```

Comment:
What's the difference between these logic checks
roads.containsKey vs !roadIsOnMap